### PR TITLE
Finish adding coverage for outstanding packages

### PR DESCRIPTION
**Pull request by fergloragain**
This PR brings the total test coverage of the project to 100%

To achieve this:

- Some publicly accessed functions were moved into vars so that we can manipulate flows in testing
- Some functions were simplified, e.g. directly returning the result of a boolean check rather than capturing the comparison result in a var and returning that, also it was possible to eliminate some unnecessary else blocks
- Some functions returned values along with an error. These functions were predominantly recursive in nature, and never actually returned a non-nil error, so the error return could be eliminated entirely
- The sequenceBreakFlag was both initialised and checked inside a loop, which was wrong, it needed to be initialised outside the loop and checked inside the loop, otherwise it would always have been tested against its initial value
- Some blocks of code were lifted out into separate functions to make them more testable
- The test data supplied for some tests was pretty nonsensical on my part, I was just feeding values in order to hit different branches of logic. I'm open to making the test data more realistic if anyone has any suggestions?
- Unfortunately I had to compress everything down into a single commit to make my rebase easier,  but hopefully I've covered most points in this PR description!

### CODE REVIEW

**Comment by TimothyStiles** [comment]
@fergloragain I'm so excited about this one. @Koeng101 has a PR (#244 ) where he fixed some Genbank Error handling. Haven't had a chance to check yet but would there be any merge conflicts?

**Comment by fergloragain** [comment]
@TimothyStiles there definitely are some conflicts with the genbank package, but I don't mind waiting for @Koeng101 to merge his changes and then I can rebase and resolve conflicts after :+1:

**Comment by fergloragain** [comment]
@TimothyStiles alternatively, I could merge this now, and then rebase @Koeng101's branch for him after?

**Comment by TimothyStiles** [comment]
> @TimothyStiles alternatively, I could merge this now, and then rebase @Koeng101's branch for him after?

@fergloragain this sounds great. Just checked and all tests pass with 100% coverage. 
Two things before merging:

1. I'll find some less messy test data for you.

2. Those functions that return values with an error that never that's non-nil were intentional. Idea being that we put some error handling in there and keep the IO syntax identical-ish across all functions.

**Comment by TimothyStiles** [comment]
@fergloragain. Just finished merging @Koeng101 genbank parser rewrite into `dev` with #244. Resolving conflicts with this PR now.

File: io/gff/gff.go
Comment (TimothyStiles) [inline io/gff/gff.go:217]: potential off-by-one error?

File: io/gff/gff.go
Comment (TimothyStiles) [inline io/gff/gff.go:238]: is this canonical in GFF?

File: io/genbank/genbank.go
Comment (TimothyStiles) [inline io/genbank/genbank.go:37]: should this be 'os.open' now?

File: io/polyjson/polyjson.go
Comment (TimothyStiles) [inline io/polyjson/polyjson.go:82]: re-add error

File: io/polyjson/polyjson.go
Comment (TimothyStiles) [inline io/polyjson/polyjson.go:87]: re-add error

File: io/polyjson/polyjson.go
Comment (TimothyStiles) [inline io/polyjson/polyjson.go:115]: linter yelling to check error

File: io/genbank/genbank.go
Comment (TimothyStiles) [inline io/genbank/genbank.go:36]: linted as dead-code

File: io/genbank/genbank.go
Comment (TimothyStiles) [inline io/genbank/genbank.go:38]: linted as dead-code

**Review by TimothyStiles — CHANGES_REQUESTED** [review]
@fergloragain this is looking really good! I've merged and cleaned up #244 and am adding back the errors even if they return nil for the moment. Any thoughts on where to go from here?

**Pull request merged**
